feature-libs/product-configurator/rulebased/core/state/effects/configurator-basic-effect.service.ts

```diff
@@ -14,35 +14,62 @@ import { Configurator } from '../../model/configurator.model';
 @Injectable({ providedIn: 'root' })
 export class ConfiguratorBasicEffectService {
   /**
-   * Finds first group with attributes for a configuration. Throws error if such a group does not exist,
-   * as this is an illegal state
+   * Finds first attribute group with attributes for a configuration (ignores conflict groups).
+   * If optional parameter 'includeConflicts' is set to true it finds first group with attributes including conflict groups.
+   * Throws error if such a group does not exist, as this is an illegal state
    * @param configuration
+   * @param includeConflicts (optional) if true it includes also conflict groups in the search
    * @returns Group id
+   *
    */
   getFirstGroupWithAttributes(
-    configuration: Configurator.Configuration
+    configuration: Configurator.Configuration,
+    includeConflicts = false
   ): string {
-    const id = this.getFirstGroupWithAttributesForList(configuration.groups);
+    const id = this.getFirstGroupWithAttributesForList(
+      configuration.groups,
+      includeConflicts
+    );
     if (id) {
       return id;
     } else {
       throw new Error('Configuration does not have any attributes');
     }
   }
+
   /**
-   * Finds first group with attributes in a list of groups
+   * Finds first group with attributes in a list of groups. Dependent on 'includeConflicts' parameters it includes conflict groups in the search or it ignores them.
    * @param groups
-   * @returns Group or undefined if such a group does not exist
+   * @param includeConflicts set to true in order to include conflict groups in the seach
+   * @returns Group id
    */
   protected getFirstGroupWithAttributesForList(
-    groups: Configurator.Group[]
+    groups: Configurator.Group[],
+    includeConflicts: boolean
   ): string | undefined {
-    const groupWithAttributes: Configurator.Group | undefined = groups
-      .filter(
-        (currentGroup) =>
-          currentGroup.attributes && currentGroup.attributes.length > 0
-      )
-      .shift();
+    let groupWithAttributes: Configurator.Group | undefined;
+    if (
+      includeConflicts &&
+      groups[0].groupType === Configurator.GroupType.CONFLICT_HEADER_GROUP
```

Comment:
There is no check that groups contains at least one record

Reply:
done